frontend/src/app/patients/EditPatient.tsx

```diff
@@ -12,6 +12,34 @@ import { LinkWithQuery } from "../commonComponents/LinkWithQuery";
 
 import PersonForm from "./Components/PersonForm";
 
+interface PatientDetails {
+  patient: {
+    firstName: string;
+    middleName: string | null;
```

Comment:
Maybe a naive question: would this PR be simpler if it just used `undefined` as its "data not set" sentinel value instead of coercing `undefined` into `null`? That would avoid any modifications to the `ComboBox` component, would allow for more succinct messages over the wire (since we have to send a `null` value, but `undefined` values are omitted), and would let you use TypeScript optionals instead of type unions.

```suggestion
    middleName?: string;
```